src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -252,26 +258,56 @@ export class DataSourceSavedObjectsClientWrapper {
 
   private async validateAttributes<T = unknown>(attributes: T) {
     const { title, endpoint, auth } = attributes;
-    if (!title?.trim?.().length) {
+
+    this.validateTitle(title);
+    await this.validateEndpoint(endpoint, attributes as DataSourceAttributes);
+    await this.validateAuth(auth);
+  }
+
+  private validateTitle(title: string) {
+    if (!title.trim().length) {
       throw SavedObjectsErrorHelpers.createBadRequestError(
         '"title" attribute must be a non-empty string'
       );
     }
 
+    if (title.length > 32) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(
+        '"title" attribute is limited to 32 characters'
+      );
+    }
+  }
+
+  private async validateEndpoint(endpoint: string, attributes: DataSourceAttributes) {
     if (!isValidURL(endpoint, this.endpointBlockedIps)) {
       throw SavedObjectsErrorHelpers.createBadRequestError(
         '"endpoint" attribute is not valid or allowed'
       );
     }
+    try {
+      const dataSourceClient: OpenSearchClient = await this.dataSourcesService.getDataSourceClient({
+        savedObjects: {} as any,
+        cryptography: this.cryptography,
+        testClientDataSourceAttr: attributes as DataSourceAttributes,
+        authRegistry: await this.authRegistryPromise,
+        customApiSchemaRegistryPromise: this.customApiSchemaRegistryPromise,
+      });
 
-    if (!auth) {
-      throw SavedObjectsErrorHelpers.createBadRequestError('"auth" attribute is required');
-    }
+      const dataSourceValidator = new DataSourceConnectionValidator(dataSourceClient, attributes);
 
-    await this.validateAuth(auth);
+      await dataSourceValidator.validate();
+    } catch (err: any) {
+      throw SavedObjectsErrorHelpers.createBadRequestError(
+        `endpoint is not valid OpenSearch endpoint`
+      );
+    }
   }
 
   private async validateAuth<T = unknown>(auth: T) {
+    if (!auth) {
+      throw SavedObjectsErrorHelpers.createBadRequestError('"auth" attribute is required');
+    }
+
```

Comment:
So from now on, we cannot choose no auth?

Reply:
creating data source with `no auth ` auth type still has an an attribute in the request

![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/32652829/04508e3f-b665-4803-a63f-2bc9bf03ca9c)